### ISSUE TITLE
string and byte conversion in events.py corrupts data

### ISSUE THREAD

**Issue by brenzi** [pipermerriam/web3.py#130]
* Version: 3.2.0
* Python: 3.5
* OS: linux

### What was wrong?
My strings passed in contract events get corrupted. They look like this when I listen to events:
`'_ipfsaddr': '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00`'`

### solution

```
--- events.py   2016-12-07 14:21:27.523182611 +0100
+++ events-new.py       2016-12-07 14:18:24.779178254 +0100
@@ -116,8 +116,8 @@
     entries using the correct types.
     """
     return [
-        'bytes32' if arg_type in {'bytes', 'string'} else arg_type
-        for arg_type in input_types
+        #'bytes32' if arg_type in {'bytes', 'string'} else arg_type
+        arg_type for arg_type in input_types
     ]
```
now I get my ipfs address correctly:
`'_ipfsaddr': 'QmSL2Zj3QmQFNcF6pethSYDF8fGYpz1LDeFiaLSzkCtAQ2'`

**Comment by 4gn3s** [comment]
I am experiencing the same behaviour with web3.py 3.2. The minimal test case:
```solidity
contract Test {
    function createEvent(string input) {
        MyEvent(input);
    }
    event MyEvent(string str);
}
```

and the test in populus:
```python
def test_event(chain):
    test_contract = chain.get_contract('Test')
    test_string = "i love food"
    event_filter = test_contract.pastEvents('MyEvent', {'toBlock': 'latest'})
    txn_hash = test_contract.transact().createEvent(test_string)
    chain.wait.for_receipt(txn_hash, timeout=120)
    event_logs = event_filter.get()[0]["args"]
    assert event_logs["str"] == test_string
```

which returns:
```python
>       assert event_logs["str"] == test_string
E       assert u'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00@'' == 'i love food'
```

**Comment by pipermerriam** [comment]
So this behavior was implemented this way under what is looking like a mis-understanding of how the logging features handle dynamic length values.  I have been under the impression that when you have a solidity event with an argument which is dynamic in length (such as a `string`) then the logged value is the `sha3` of that value.

* I'm pretty sure this is true if the argument is `indexed` because topics are always 32 bytes.

Can either of you confirm (with something like geth or parity) that when you log a non-indexed `string` that the full value is present in the `data` portion of the log?  I'm unable to verify myself right now due to destroying my primary work laptop with a glass of water.

**Comment by 4gn3s** [comment]
According to this: https://solidity.readthedocs.io/en/latest/contracts.html?highlight=log#events `All non-indexed arguments will be stored in the data part of the log.` I have checked my contract in browser-solidity and it shows the full string in the event. I'm not on my main machine right now either, and can check it on Monday, sorry.

**Comment by pipermerriam** [comment]
@4gn3s Thanks.  Classifying this as a bug.  I believe the fix should be pretty easy.  Hope to get something out today but no promises.

**Comment by 4gn3s** [comment]
Finally had a chance to test this in geth. It also returns `{u'str': u'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00 '}`. Am I missing something here, or is it a bug as well? There is an example in the ethereum wiki: https://github.com/ethereum/wiki/wiki/Ethereum-Contract-ABI#example-javascript-usage and I think it also suggests that the data should be returned as-is. Unfortunately cannot check parity right now.

**Comment by pipermerriam** [comment]
This issue should be fixed in releases  `>=3.4.3`
